git/operations.go

```diff
@@ -305,6 +339,11 @@ func execGitCmd(ctx context.Context, args []string, config gitCmdConfig) error {
 			err = errors.New(msg)
 		}
 	}
+
+	if trace && !exemptedTraceGitCommand(args[0]) {
+		println(traceGitCommand(args, config, traceOutput.String()))
```

Comment:
I would just inline the yes/no logic of `exemptedTraceGitCommand`, `traceGitCommand` and the `println` into one func, and call that here. That would remove some noise from this func.

Reply:
I considered doing that but the result of that could mean that someone would set `trace` to `true` and then see:

```go
if trace {
  traceGitCommand(...)
}
```
but in fact that command could full well do nothing.

On second thought, since (as it stands now, but not as it was then when I made that decision) there's nothing in the exempted list, then I think it makes sense to assume that if someone added something to that list (temporarily) then that person will fully understand what's going on.

I'll make that change.

Reply:
as for inlining the println.. how would I then test that?  otherwise I would need to do

```go
		traceCommand = traceGitCommand(args, config, traceOutput.String())
		if traceCommand != "" {
			println(traceCommand)
		}
```

Reply:
You can keep `traceGitCommand` factored out to be able to test it. I was more concerned about `exemptedGitCommand` standing on its own -- it's too simple to justify factoring it out, but distracting when inlined into `execGitCmd` directly.

Reply:
ok, I will keep it like this then:

```
	if trace {
		traceCommand := traceGitCommand(
			args,
			config,
			traceStdout.String(),
			traceStderr.String(),
		)
		if traceCommand != "" {
			println(traceCommand)
		}
	}
```

Reply:
That seems reasonable yup. You could shorten that a little to

```
if trace {
		if traceCommand := traceGitCommand(
			args,
			config,
			traceStdout.String(),
			traceStderr.String(),
		); traceCommand != "" {
			println(traceCommand)
		}
	}
```

(also there's no need to have each arg on its own line -- for better or worse, it's not the Go Way to limit line length (or at least, not to limit it to 72 chars ..))